public/app/features/dashboard/dashgrid/DashboardGrid.tsx

```diff
@@ -152,10 +152,15 @@ export class DashboardGrid extends PureComponent<Props> {
     for (const panel of this.props.dashboard.panels) {
       const panelClasses = classNames({ 'react-grid-item--fullscreen': panel.isViewing });
 
+      // sets a descending z-index to each panel to allow overflowing content to show on top of the next panel
+      const panelZIndex = this.props.dashboard.panels.length - panelElements.length;
+      const panelStyles = { zIndex: panelZIndex };
+
```

Comment:
can you move this logic to https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/dashgrid/DashboardGrid.tsx#L295 instead? feels like it belongs better inside GrafanaGridItem where other grid item styling.

Reply:
moved the style setting to the item, but the calculation of the descending order is better in the grid itself